src/test/groovy/com/streamr/client/utils/MessageCreationUtilSpec.groovy

```diff
@@ -138,33 +145,22 @@ class MessageCreationUtilSpec extends Specification {
     }
 
     void "creates encrypted messages when key defined in constructor"() {
-        String key = genKey(32)
-        MessageCreationUtil util = new MessageCreationUtil("publisherId", null, [(stream.id): key])
+        GroupKey key = genKey(32)
+        keyStorage.addKey(stream.id, key)
+        MessageCreationUtil util = new MessageCreationUtil("publisherId", null, keyStorage)
         when:
         StreamMessageV31 msg = (StreamMessageV31) util.createStreamMessage(stream, message, new Date(), null)
         then:
         assert msg.encryptionType == StreamMessage.EncryptionType.AES
         assert msg.getSerializedContent().length() == 58 // 16*2 + 13*2 (hex string made of IV + msg of 13 chars)
     }
 
-    void "throws if the key is not 256 bits long"() {
-        String key = genKey(16)
-        when:
-        new MessageCreationUtil("publisherId", null, [(stream.id): key])
-        then:
-        thrown InvalidGroupKeyException
-
-        when:
-        msgCreationUtil.createStreamMessage(stream, message, new Date(), null, key)
-        then:
-        thrown InvalidGroupKeyException
-    }
-
     void "creates encrypted messages when key defined in createStreamMessage() and use the same key later"() {
```

Comment:
You're right, I added these checks under `EncryptionUtilSpec`. But I also left them here as an additional precaution in case `createStreamMessage()` is later modified and by inadvertence doesn't call `encrypt()` properly.